pkgs/development/compilers/ghc/cabal-paths-9.2.x.patch

```diff
@@ -0,0 +1,278 @@
+diff --git a/Cabal/src/Distribution/Simple/Build/PathsModule.hs b/Cabal/src/Distribution/Simple/Build/PathsModule.hs
+index b2be7e1a8..bf218dbc1 100644
+--- a/Cabal/src/Distribution/Simple/Build/PathsModule.hs
++++ b/Cabal/src/Distribution/Simple/Build/PathsModule.hs
+@@ -25,16 +25,21 @@ import Distribution.Package
+ import Distribution.PackageDescription
+ import Distribution.Simple.Compiler
+ import Distribution.Simple.LocalBuildInfo
+-import Distribution.Simple.Utils          (shortRelativePath)
++import Distribution.Simple.Utils          (shortRelativePath, unintersperse)
+ import Distribution.System
+ import Distribution.Version
+ 
+ import qualified Distribution.Simple.Build.PathsModule.Z as Z
+ 
++import System.FilePath ( pathSeparator )
++
+ -- ------------------------------------------------------------
+ -- * Building Paths_<pkg>.hs
+ -- ------------------------------------------------------------
+ 
++splitPath :: FilePath -> [ String ]
++splitPath = unintersperse pathSeparator
++
+ generatePathsModule :: PackageDescription -> LocalBuildInfo -> ComponentLocalBuildInfo -> String
+ generatePathsModule pkg_descr lbi clbi = Z.render Z.Z
+     { Z.zPackageName                = packageName pkg_descr
+@@ -56,8 +61,41 @@ generatePathsModule pkg_descr lbi clbi = Z.render Z.Z
+     , Z.zDatadir    = zDatadir
+     , Z.zLibexecdir = zLibexecdir
+     , Z.zSysconfdir = zSysconfdir
++
++    , Z.zAbsBody = absBody
++    , Z.zWarnPragma = warnPragma
++    , Z.zImportList = importList
++    , Z.zShouldEmitDataDir = shouldEmitDataDir
+     }
+   where
++    dirs = [ (flat_libdir, "LibDir")
++           , (flat_dynlibdir, "DynLibDir")
++           , (flat_datadir, "DataDir")
++           , (flat_libexecdir, "LibexecDir")
++           , (flat_sysconfdir, "SysconfDir") ];
++    shouldEmitPath p
++      | (splitPath flat_prefix) `isPrefixOf` (splitPath flat_bindir) = True
++      | (splitPath flat_prefix) `isPrefixOf` (splitPath p) = False
```

Comment:
@sternenseemann Continuing the review from the last PR, what do you think is the proper fix here? I can't work out why splitPath was included originally (I copied this from the existing cabal-paths.patch file), but I imagine the correct code would be one of the following:
```suggestion
+      | flat_prefix `isPrefixOf` flat_bindir = True
+      | flat_prefix `isPrefixOf` p = False
```
```suggestion
+      | [flat_prefix] `isPrefixOf` (splitPath flat_bindir) = True
+      | [flat_prefix] `isPrefixOf` (splitPath p) = False
```

Reply:
Yes, I think so.

Reply:
Resolved this with a variation on the top patch, which gets rid of splitPath entirely.